File: src/main/scala/interviewbit/string/LastWordLength.scala
package interviewbit.string

class LastWordLength {
  def lengthOfLastWord(s: String): Int  =
    s.foldRight(0, false) { case (c, (len, inWord)) =>
      if (c == ' ' && inWord) return len
      else if (c == ' ') (len, false)
      else if (inWord) (len + 1, true)
      else (1, true)
    }._1
}

File: src/main/java/edu/ren/datastructure/interviewBit/string/LastWordLength.java
package edu.ren.datastructure.interviewBit.string;


public class LastWordLength {
    public static int lengthOf1(String inputStr) {
        String input = inputStr.trim();
        String[] split = input.split(" ");
        String lastWord = split[split.length - 1];
        return lastWord.length();
    }

    public static int lengthOf(String inputStr) {
        int length = 0;
        for (int i = inputStr.length() - 1; i >= 0; i--) {
            if (inputStr.charAt(i) != ' ') {
                length++;
                if (i > 0 && inputStr.charAt(i - 1) == ' ') {
                    break;
                }
            }
            /* else {//current char is space and right is letter
                if (i < inputStr.length()-1 && inputStr.charAt(i + 1) != ' ') {
                    break;
                }*/
        }
        return length;
    }
}
